### ISSUE TITLE
Warning with clone

### ISSUE THREAD

**Issue by Yash-Vardhan-Sharma** [learnables/learn2learn#188]
/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py:385: UserWarning: The .grad attribute of a Tensor that is not a leaf Tensor is being accessed. Its .grad attribute won't be populated during autograd.backward(). If you indeed want the gradient for a non-leaf Tensor, use .retain_grad() on the non-leaf Tensor. If you access the non-leaf Tensor by mistake, make sure you access the leaf Tensor instead. See github.com/pytorch/pytorch/pull/30531 for more informations.
  if param.grad is not None:

This is the warning i get everytime , even with miniimagenet maml code of your repo , This is associated with maml.clone() thing .
Can u put light on what this is ?

And also I am training a resnet18 (5way) with softmax and crossentropy loss , my loss is getting saturated at ln(5) ie ~1.62 from iteration 1 only . I checked the maml.parameters.grad values , they are non zero but quite low (~1e-6) .
For this till now i have tried changing softmax to sigmoid activation and using different optimizers sgd,adam,rmsprop .
Problem is still there, can u put light on this too .
i tried using prototypical network for same network training and that is working atleast it is getting trained .

Thanks

**Comment by Yash-Vardhan-Sharma** [comment]
Thanks for the changes , any comments on my other issues ?

**Comment by seba-1511** [comment]
Not sure I can help, MAML is known to be quite finicky to train with deeper networks. I'd recommend fewer adaptation steps and small fast adaptation learning rates. ProtoNet or ANIL are likely safer bets for a ResNet18.
